libvips/resample/reduceh.cpp

```diff
@@ -435,22 +438,61 @@ vips_reduceh_build( VipsObject *object )
 	VipsResample *resample = VIPS_RESAMPLE( object );
 	VipsReduceh *reduceh = (VipsReduceh *) object;
 	VipsImage **t = (VipsImage **) 
-		vips_object_local_array( object, 2 );
+		vips_object_local_array( object, 3 );
 
 	VipsImage *in;
-	double width, extra_pixels;
+	int width;
+	int int_hshrink;
+	double extra_pixels;
 
 	if( VIPS_OBJECT_CLASS( vips_reduceh_parent_class )->build( object ) )
 		return( -1 );
 
 	in = resample->in; 
 
-	if( reduceh->hshrink < 1 ) { 
+	if( reduceh->hshrink < 1.0 ) { 
 		vips_error( object_class->nickname, 
-			"%s", _( "reduce factors should be >= 1" ) );
+			"%s", _( "reduce factor should be >= 1.0" ) );
 		return( -1 );
 	}
 
+	/* Output size. We need to always round to nearest, so round(), not
+	 * rint().
+	 */
+	width = VIPS_ROUND_UINT(
+		(double) in->Xsize / reduceh->hshrink );
+
+	/* How many pixels we are inventing in the input, -ve for
+	 * discarding.
+	 */
+	extra_pixels = width * reduceh->hshrink - in->Xsize;
+
+	if( reduceh->gap != 0.0 &&
```

Comment:
Testing equality on doubles is always a bit wonky. Maybe `> 0.0` is slightly cleaner?

Reply:
Indeed, resolved with commit 1282790529b1bf8cf5af26da431e3746b4be21c3.